lib/Doctrine/DBAL/Connection.php

```diff
@@ -646,23 +650,31 @@ public function getTransactionIsolation()
      */
     public function update($tableExpression, array $data, array $identifier, array $types = array())
     {
+        $columnList = array();
         $set = array();
+        $criteria = array();
+        $paramValues = array();
 
         foreach ($data as $columnName => $value) {
+            $columnList[] = $columnName;
             $set[] = $columnName . ' = ?';
+            $paramValues[] = $value;
         }
 
-        if (is_string(key($types))) {
-            $types = $this->extractTypeValues(array_merge($data, $identifier), $types);
+        foreach ($identifier as $columnName => $value) {
```

Comment:
This `foreach` matches 1:1 the previous one. Same goes for the  subsequent one, besides for one line diff. Is it possible to give this a name, and throw it into a private method?